src/Akeneo/Category/back/Infrastructure/Controller/InternalApi/ImagePreviewController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Infrastructure\Controller\InternalApi;
+
+use Akeneo\Category\Infrastructure\FileSystem\PreviewGenerator\CouldNotGeneratePreviewException;
+use Akeneo\Category\Infrastructure\FileSystem\PreviewGenerator\PreviewGeneratorInterface;
+use Akeneo\Category\Infrastructure\Storage\InMemory\GetAttributeInMemory;
+use Liip\ImagineBundle\Binary\Loader\LoaderInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\HttpKernel\EventListener\AbstractSessionListener;
+
+/**
+ * Fetches the binary preview of the image
+ *
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class ImagePreviewController
+{
+    private const THUMBNAIL_FILENAME = 'thumbnail.jpeg';
+
+    /**
+     * Why do we need to have this?
+     * At first this action was supposed to redirect to the final media. As we now use flysystem to store
+     * our thumbnails and the storage is not public. We cannot directly redirect to a public URL.
+     * Unfortunatly, liipimagine is designed to directly provide a file URL. So we twisted a bit the system
+     * to get the flysystem path instead of a public URL. As liipimagine requires to define a web_root
+     * (cannot be empty), we decided to put a dummy root flag (here: vendor/akeneo/pim-community-dev/config/packages/liip_imagine.yml#L8)
+     * and remove it here to get the flysystem path.
+     */
+    private const ROOT_FLAG = '__root__';
+
+    public function __construct(
+        private GetAttributeInMemory $getAttributeInMemory,
```

Comment:
We should use the `GetAttribute` interface instead of the `InMemory` implementation. The implementation will be injected into the dependency configuration
```suggestion
        private GetAttribute $getAttribute
```